### ISSUE TITLE
F36: CHANGE: Retired Packages

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#1099]
In the [2022-02-09 community meeting](https://lists.fedoraproject.org/archives/list/coreos@lists.fedoraproject.org/message/E47SUSGIWAUP77HK2SOVDJQRT7HPAH4E/) we decided to discuss further the following change:

- [Retired Packages](https://fedoraproject.org/wiki/Changes/RetiredPackages)
    - Ease the task of removing packages, which were retired and no longer receives updates.
    - Tracking bug: [#2018192](https://bugzilla.redhat.com/show_bug.cgi?id=2018192)         

This change is more about yum based systems having a tool that will detect and remove retired packages when doing a major upgrade. While it doesn't apply directly to FCOS we decided to open this issue to further discussion around our own particular problem.

We believe that if a user has layered a package that then is retired and no longer exists it will yield them into a state where they won't be able to auto update (the upgrade will fail). Their auto-updating system is now stuck until user intervention is applied. We should discuss how to handle this.

**Comment by jlebon** [comment]
This is similar to https://github.com/coreos/rpm-ostree/issues/3465. IMO, I think here too the right behaviour is to keep erroring out, but possibly provide a better error message.

**Comment by jlebon** [comment]
Closely related to this is the subject of monitoring and alerting. Zincati already has a Prometheus interface through which it could report these error conditions.

**Comment by dustymabe** [comment]
> This is similar to [coreos/rpm-ostree#3465](https://github.com/coreos/rpm-ostree/issues/3465). IMO, I think here too the right behaviour is to keep erroring out, but possibly provide a better error message.

Agree, but I think there was one further idea: https://github.com/coreos/console-login-helper-messages/issues/107
